src/ddsrt/include/dds/ddsrt/time.h

```diff
@@ -59,6 +59,10 @@ typedef int64_t dds_duration_t;
 /** @name Infinite timeout for relative time */
 #define DDS_INFINITY ((dds_duration_t) INT64_MAX)
 
+/** @name Invalid time */
+#define DDS_TIME_INVALID ((dds_time_t) 0xffffffffffffffffull)
+
```

Comment:
Is it necessary to introduce DDS_TIME_INVALID? Can't you use DDS_NEVER? If DDS_TIME_INVALID is needed, then you probably have to update ddsrt at various places to be able to cope with that.

Reply:
I've changed to DDS_NEVER

Reply:
But ``NEVER`` is something fundamentally different from ``INVALID``: in this code base, all time stamps will be less than ``NEVER`` — the logical consequence is that you have to stop execution before Friday the 11th of April 2262 at 23:47:16 UTC (give or take a few nanoseconds, and I'm not quite sure how many leap seconds will be introduced) — but ``NEVER`` is certainly a valid value in many places where a timestamp is required, e.g., for scheduling events.

I would not expect to be allowed to schedule an event for an ``INVALID`` time. I don't know how you intend to use ``INVALID`` and whether this distinction actually matters. It is quite possible that using ``NEVER`` instead is perfectly fine, but it is not self-evident.

Defining ``INVALID`` as ``INT64_MIN`` seems like a much more logical choice.

Reply:
defined DDS_INVALID